### ISSUE TITLE
Low Discrepancy Sequences for MonteCarlo and VEGAS

### ISSUE THREAD

**Issue by FHof** [esa/torchquad#140]
# Feature
## Desired Behavior / Functionality

Low discrepancy sequences could be used as an optional replacement for random number generation for MonteCarlo and VEGAS.
With certain integrands, this could lead to a higher average accuracy. 

## What Needs to Be Done

* Add a class similar to the RNG class to generate numbers with a low discrepancy sequence instead of a PRNG. An instance of it can be passed as `rng` argument to VEGAS and MonteCarlo.
  This class could use, for example, PyTorch's and TensorFlow's sobol sequences: https://www.tensorflow.org/api_docs/python/tf/math/sobol_sample, https://pytorch.org/docs/stable/generated/torch.quasirandom.SobolEngine.html
* Add a function to the number generator classes which samples points and use it for MonteCarlo and VEGAS instead of `uniform`. In comparison to `uniform`, the output of this function always corresponds to points in a space, where conceptually a distance function is defined.
* Change `MonteCarlo.get_jit_compiled_integrate` so that it works with the number generator class for low discrepancy sequences.

## How Can It Be Tested

It can be tested with additional tests in the torchquad/tests folder.

**Comment by Samuel-Vangu** [comment]
Hi @gomezzz, nice to see Sobol added in #244! I'd be interested in extending this to
other low-discrepancy sequences (Halton, etc.) — I have prior experience with QMC methods
(Sobol, Halton, LHS, Korobov lattices,etc) and recently contributed several discrete
distribution classes to QMCPy. Could I be assigned to work on this?

**Comment by gomezzz** [comment]
Implemented in [v0.6.0](https://github.com/esa/torchquad/releases/tag/v0.6.0).

`Sobol` is a scrambled low-discrepancy sampler shaped like `RNG`, so it drops straight into the `rng` argument:

```python
from torchquad import MonteCarlo, Sobol
MonteCarlo().integrate(fn, dim=2, N=2**16, integration_domain=dom, rng=Sobol("torch", seed=0))
```

It uses `torch.quasirandom.SobolEngine` on the torch backend and `scipy.stats.qmc.Sobol` elsewhere, pushing the error toward `O(1/N)` rather than `O(1/sqrt(N))` on smooth integrands. Sample points remain constants, so gradients still flow through the integrand and the domain. Verified against the published wheel: on `sin(x)` over `[0, pi]` at `N=2**16`, plain Monte Carlo lands at ~3e-3 absolute error while Sobol is at machine precision.
